include/dynamic-graph/signal-time-dependent.h

```diff
@@ -13,16 +13,51 @@ namespace dynamicgraph {
   signals,
   making sure its inputs are up to date on access, using a incrementing time
   tick as reference.
-  It works this way: for a given SignalTimeDependent S, the user manually
-  adds dependent signals through the
-  use of the addDependency function. On access (calling the signal S
-  operator  () or access(Time) function),
-  if the dependent signals are not up-to-date, i.e. if their [last update]
-  time is less than the
-  current time, their value will be access ()'ed to bring them up-to-date.
-  Thus, the value of dependent
-  signals can be accessed \b quickly and \b repeatedly through the
-  accessCopy () function.
+
+  It works this way. For a given SignalTimeDependent S,
+  \li the user manually adds dependent signals through the use of the SignalTimeDependent::addDependency function.
+  \li On access (calling the signal S SignalTimeDependent::operator() or
+  SignalTimeDependent::access(Time) function), if the dependent signals are not
+  up-to-date, i.e. if their [last update] time is less than the current time,
+  their value will be SignalTimeDependent::access ()'ed to bring them up-to-date.
+
+  Thus, the value of dependent signals can be accessed \b quickly and
+  \b repeatedly through the Signal::accessCopy () function.
+
+  An example:
+  \code
+  class MyEntity : public Entity {
+  public:
+  // Some signal dependencies
+  SignalPtr<T,int> dep1, dep2;
+  SignalTimeDependent<T,int> signal;
+
+  MyEntity (const std::string& name)
+    : Entity (name)
+    , signal (
+        // Set the function that computes the signal value
+        boost::bind (&Entity::computeSignal, this, _1, _2),
+        // Declare the dependencies
+        dep1 << dep2,
+        "signalname")
+  {}
+
+   T& computeSignal (T& res, int time)
+   {
+     // dep1 and dep2 are already up-to-date.
+     // You should access them as
+     dep1.accessCopy();
+     dep2.accessCopy();
+     // Accessing them as below is less efficient because it will check that the
+     // signal is up-to-date, while you know it is.
+     // dep1(time);
+     // dep1.access(time);
+     // dep1.recompute(time);
```

Comment:
This seems to be wrong. @olivier-stasse could you confirm that even though dep1 and dep2 are dependencies of signal, they are not recomputed automatically ?
My understanding of the unit-test says they aren't recomputed automatically.

Reply:
Firstly according to https://github.com/stack-of-tasks/dynamic-graph/blob/1bcf9a54b7c3ea75d9230a15f25df92af28cbcd5/include/dynamic-graph/signal.h#L113 dep1(time) is equivalent to dep1.access(time)
Secondly according to https://github.com/stack-of-tasks/dynamic-graph/blob/1bcf9a54b7c3ea75d9230a15f25df92af28cbcd5/include/dynamic-graph/signal.h#L107 dep1.recompute(time) is equivalent to dep1.access(time)

As dep1 and dep2 are not time dependent, the time given in the signal is just an information about when was the last call to access(). If call accessCopy you will indeed stay with the copy of the last call, but that will not be updated if needed. The signal-time-dependent reimplement access() and check if it needs to be updated:

-  if yes it calls the method access inherited, providing a real access and store the value. 
-  if no it just returns accessCopy()
I think therefore that you should use access(time) and not accessCopy(time), as it does what you need

see https://github.com/stack-of-tasks/dynamic-graph/blob/1bcf9a54b7c3ea75d9230a15f25df92af28cbcd5/include/dynamic-graph/signal-time-dependent.h#L91 for the reimplementation of access.

I did not check yet if there is a good unit-test testing this.
I will try to write one if not, otherwise we may create a link to it in the documentation.